heapster-saw/doc/tutorial/tutorial.md

```diff
@@ -0,0 +1,623 @@
+# Tutorial to learn the basics of heapster-saw
+
+This tutorial extends the current README with enough details and
+examples to get anyone up to speed with using and hacking on Heapster.
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [Tutorial to learn the basics of heapster-saw](#tutorial-to-learn-the-basics-of-heapster-saw)
+    - [Building](#building)
+        - [Build Saw](#build-saw)
+        - [Build the Coq backend for Saw](#build-the-coq-backend-for-saw)
+        - [Build all the examples](#build-all-the-examples)
+    - [A quick tour of SAW](#a-quick-tour-of-saw)
+        - [Overview](#overview)
+        - [Running an example](#running-an-example)
+        - [Batch scripts](#batch-scripts)
+    - [Using Heapster](#using-heapster)
+        - [Heapster type-checking overview](#heapster-type-checking-overview)
+        - [Running an example](#running-an-example-1)
+    - [Looking under the hood](#looking-under-the-hood)
+        - [Heapster commands and environments](#heapster-commands-and-environments)
+        - [Permissions](#permissions)
+        - [Type-checking](#type-checking)
+
+<!-- markdown-toc end -->
+
+
+## Building
+
+We'll start by building everything you need to use Heapster. All the
+commands here are with respect to the top-level saw-script directory.
+
+### Build Saw 
+
+You will need to follow the instructions in the top-level README to
+download or build a SAW binary, of which Heapster is a part. In
+particular, make sure you follow the instructions to install Z3. Once `./build.sh`
+succeeds it should report 
+
+```bash
+COPIED EXECUTABLES TO /Path/To/Saw/saw-script/bin.
+```
+
+If everything is installed correctly you should be able to run saw
+
+```bash
+cabal run saw
+```
+
+This should open the saw interactive session. It should show you a
+pretty SAW logo with the version number. We will learn more about the
+interactive session later. For now you can quit the session with `:quit` or
+`:q` like so:
+
+```bash
+sawscript> :quit
+```
+
+### Build the Coq backend for Saw
+
+In this tutorial we will also interact with Heapster's Coq output. So you'll need
+to follow the instructions in the README in the `saw-core-coq` subdirectory.
```

Comment:
Please add a link to that README

Reply:
What do you have in mind for this link? Do you mean a hyperlink to the README in the master branch in github?

The location would be different if the tutorial is being read online or locally. That's why I tend to use local commands when I can.